server/modules/google-analytics.js

```diff
@@ -0,0 +1,11 @@
+let GoogleAnalytics = (() => {
+    function getGoogleAnalyticsScript(tagId) {
+        return "<script async src=\"https://www.googletagmanager.com/gtag/js?id=" + tagId + "\"></script>" +
+            "<script>window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date());gtag('config', '" + tagId + "'); </script>";
```

Comment:
How is tagId sanitized when it is input. Is there anything stopping a user from entering something like 
```
<valid google tag>'); alert('Gotcha
```
or other, more questionable input?

Reply:
I have now added a regex check to validate tags

Reply:
This is a XXS issue. It must be escaped on the rendering side, because if people reuse this function in the future without knowing it is vulnerable, it can be very dangerous.

I think [jsece](https://github.com/mathiasbynens/jsesc) should help to escape.

Reply:
I've now added jsesc when rendering the script.